dlib/simd/simd8f.h

```diff
@@ -3,11 +3,9 @@
 #ifndef DLIB_sIMD8F_Hh_
 #define DLIB_sIMD8F_Hh_
 
-#include "simd_check.h"
```

Comment:
I think the easiest course is for something to include dlib/simd.h,    it includes the checks and picks the appropriate version.   It makes the most sense,  and can be used on multiple architectures. I don't think it's going to hurt anything if it's included,  the ifdef in the header will keep it from being included twice.